documentation/rest-clients.md

```diff
@@ -0,0 +1,158 @@
+# Typescript REST Clients
+Our REST Clients provide simple, flexible, and reliable connections to REST APIs. Although these clients are close to the REST layer, they provide benefits such as a strongly typed experience that helps to navigate the REST API through auto-complete and IntelliSense. We also provide some functionality out of the box such as retry handling and authentication.
```

Comment:
> Although these clients are close to the REST layer, they provide benefits such as a strongly typed experience that helps to navigate the REST API through auto-complete and IntelliSense

I sort of wonder if people will get the wrong idea about what's going on when we say "close to the REST layer".  At the end of the day (TypeScript Typing aside) these clients provide raw access to the REST API of the service.  For users of typescript, we also provide a set of type hints to enable rich type checking and to drive intelligence, but that's all just sugar and doesn't move them "further away from the rest layer".

Reply:
> We also provide some functionality out of the box such as retry handling and authentication.

I don't know what (if any) collateral we have about the pipeline for JavaScript, but if we do have some we might link to it here.  My understanding is that the rest clients and higher level clients share the same basic pipeline model and policies, I'm not sure if customers would find that interesting or not.

Reply:
I believe making the point of "you better use this instead of raw RestAPI call" is fair advertising, though I get the idea of not making it too verbose as well. I feel we're in the right spot, especially with authentication. I could not really make a case for retries for instance in Python, since default HTTP lib do retries anyway, so you don't need us for that, while authentication you REALLY need us :)

Reply:
Thanks for the input! Really helpful 😄 

I have updated the text to be less verbose about the out-of-the-box features and linked them to the pipeline docs we have.